### ISSUE TITLE
Kompose doesn't work with docker-compose v3.7

### ISSUE THREAD

**Issue by dexterarun** [kubernetes/kompose#1152]
When I try to convert a docker-compose file v3.7 to the various files required by kubernetes using Kompose v1.18.0 , I get the message: Version 3.7 of Docker Compose is not supported. Please use version 1, 2 or 3. Isn't it time yet to update the supported versions?

**Comment by Clement-TS** [comment]
Same here with 3.5.
Some useful docker-compose parameters such as `start_period` and network `name` were added since Version 3...

**Comment by hangyan** [comment]
Yes. Thanks for create this issue, i will create a PR to add this.

**Comment by KhojiZoda** [comment]
Hello, saw your issue today, I was just wondering if the 3.7 version is not supported, is the 3.6 version supported?

**Comment by hangyan** [comment]
wired. I have check the source code and both 3.6 and 3.7 is supported, can you try the master build?

**Comment by KhojiZoda** [comment]
What version of kompose do you have?

**Comment by crazyberry7** [comment]
Same issue. "Version 3.7 of Docker Compose is not supported. Please use version 1, 2 or 3"

**Comment by jdarling** [comment]
Get the same thing with 1.17.0

```bash
./kompose version
1.17.0 (a74acad)

./kompose convert
FATA Version 3.7 of Docker Compose is not supported. Please use version 1, 2 or 3 
```

If I change the version in the docker-compose.yml file to '3' instead of '3.7' everything works fine, so it's "supported" but doesn't want to be validated LOL

**Comment by yjkimjunior** [comment]
bump? I'm able to `kompose convert` without any trouble but `kompose up` says 
```
Error while deploying application: k.Transform failed: image key required within build parameters in order to build and push service
```
The image key is populated already from the `kompose convert` command from what I can see. I've also tried setting it to the remote endpoint where it is hosted (on Google Container Registry) to the same result. 

My `docker-compose` looks more or less like:
```
version: '3.7'
services:
  main:
    build: ./
  postgres:
    image: postgres:12.1
    restart: always
    ports:
      - '5432:5432'
    environment:
      POSTGRES_USER: ...
      POSTGRES_PASSWORD: ...
    volumes:
      - postgres:/var/lib/postgresql/data
  prisma:
    image: prismagraphql/prisma:1.34
    restart: always
    ports:
      - '4466:4466'
    environment:
      PRISMA_CONFIG: |
        port: 4466
        databases:
          default:
            connector: postgres
            host: postgres
            user: ...
            password: ...
            rawAccess: true
            port: 5432
            migrations: true
volumes:
  postgres: ~

```

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  annotations:
    kompose.cmd: kompose convert
    kompose.version: 1.19.0 ()
  creationTimestamp: null
  labels:
    io.kompose.service: main
  name: main
spec:
  replicas: 1
  strategy: {}
  template:
    metadata:
      annotations:
        kompose.cmd: kompose convert
        kompose.version: 1.19.0 ()
      creationTimestamp: null
      labels:
        io.kompose.service: main
    spec:
      containers:
      - image: main
        name: main
        resources: {}
      restartPolicy: Always
status: {}

```

**Comment by alechp** [comment]
@yjkimjunior https://github.com/kubernetes/kompose/issues/969 has a partial solution

TLDR: Add image name to your prisma service

**Before including image name**
![Screen Shot 2019-12-31 at 10 57 15 AM](https://user-images.githubusercontent.com/3058939/71631464-a9743500-2bbd-11ea-82b7-742d911cf031.png)

**After including image name**
![image](https://user-images.githubusercontent.com/3058939/71631471-b2fd9d00-2bbd-11ea-8de4-4f3e151a42d6.png)

![Screen Shot 2019-12-31 at 11 05 39 AM](https://user-images.githubusercontent.com/3058939/71631483-bbee6e80-2bbd-11ea-97a0-41eb41b8efb4.png)

**Comment by hangyan** [comment]
@alechp Thanks for the explanation, i have merged PR #1213 to solve this empty image key issue, it will be included in the `1.21.0` release.

**Comment by yjkimjunior** [comment]
@alechp thanks! I decided to just write my kubernetes config files manually for current project but hopefully will revisit kompose in the near future.

**Comment by moluzhui** [comment]
@hangyan The kompose conver command failed to execute. The output is similar to the following
```
# docker-compose config > docker-compose-resolved.yaml
# kompose convert -f docker-compose-resolved.yaml
FATA Version 2.4 of Docker Compose is not supported. Please use version 1, 2 or 3
```
Official documentation says restrictions support `2.1` and `3.2`, why is 2.4 not supported?

**Comment by lebadapetru** [comment]
I have kompose version 1.21.0 (992df58d8) on win 10 pro and i still receive: _?[31mFATA?[0m Version 3.8 of Docker Compose is not supported. Please use version 1, 2 or 3_

**Comment by hangyan** [comment]
@lebadapetru minor version of `1` `2` is not supported, docker-compose 3 is supported to version `3.7`

**Comment by kachar** [comment]
Issue seems to be still ongoing with the latest release tag `1.21` from 25 Feb 2020

```
$ kompose version
1.21.0 (992df58d8)
```

With `docker-compose.yaml` version `3.8`

```yaml
version: "3.8"
services:
...
```

Do you plan a new release with the adoption with granular versions like `3.8`?

Any workaround is also welcome

**Comment by atkuzmanov** [comment]
Hi @hangyan ,

I just tried the Kompose master build v1.22.0 and I can confirm that it is working for Docker-Compose v3.8.
Thanks for the great work!

@kachar, it should work for you now, unless you have some healthcheck or other deprecated specifics in depends_on, please see:
<https://stackoverflow.com/questions/47710767/what-is-the-alternative-to-condition-form-of-depends-on-in-docker-compose-versio>

Cheers,
atkuzmanov

**Comment by rodney-b** [comment]
I'm still having this issue with the latest Kompose Linux release version v1.22.0 (955b781). If I try to `kompose convert` with the version in my docker compose yaml set to "3.8" I get:

```
FATA Version 3.8 of Docker Compose is not supported. Please use version 1, 2 or 3
```

**Comment by hangyan** [comment]
@rodney-b I think it's may be in the master build, you can try it or wait for the next release

**Comment by mooreInteractive** [comment]
I had "3.9" just changed to "3" and convert worked. Not sure the implication, but seems to have worked.

**Comment by manatax** [comment]
Getting it on 3.9 with version 1.24.0

**Comment by ons-cherif** [comment]
UPDATES in 2022: It doesn't work for me with 3.9, but it works fine with 3.6 and 3.7.

**Comment by boturkin** [comment]
I have version 1.26.0 (40646f47). It doesn't work with 3.9

**Comment by SteversIO** [comment]
Tags property is not supported in `kompose` either.

FATA services.api.build Additional property tags is not allowed

**Comment by cdrage** [comment]
Honestly we need to update Kompose to take any file that's above version 3. 

Especially because the Compose spec has removed the version key. 

We'll push a new PR soon!

**Comment by SterlingMcKinley** [comment]
kompose convert worked for me after changing the version of my docker-compose.yml file to "3".

Once I did that kompose convert worked with no issue.
